src/features/lightspeed/handleApiError.ts

```diff
@@ -1,25 +1,40 @@
 import { AxiosError } from "axios";
+import {
+  ERRORS,
+  ERRORS_UNAUTHORIZED,
+  ERRORS_TOO_MANY_REQUESTS,
+  ERRORS_BAD_REQUEST,
+  ERRORS_UNKNOWN,
+  ERRORS_CONNECTION_TIMEOUT,
+} from "./errors";
 
 export function retrieveError(err: AxiosError): string {
   if (err && "response" in err) {
-    if (err?.response?.status === 401) {
-      return "User not authorized to access Ansible Lightspeed.";
-    } else if (err?.response?.status === 429) {
-      return "Too many requests to Ansible Lightspeed. Please try again after some time.";
-    } else if (err?.response?.status === 400) {
-      const responseErrorData = <AxiosError<{ message?: string }>>(
-        err?.response?.data
-      );
-      if (
-        responseErrorData &&
-        responseErrorData.hasOwnProperty("message") &&
-        responseErrorData.message?.includes("Cloudflare")
-      ) {
-        return `Cloudflare rejected the request. Please contact your administrator.`;
-      } else {
-        return "Bad Request response. Please try again.";
-      }
-    } else if (err?.response?.status === 403) {
+    const responseErrorData = <AxiosError<{ message?: string }>>(
+      err?.response?.data
+    );
+    // Lookup _known_ errors
+    const status: number = err?.response?.status ?? 500;
+    const code: string = responseErrorData.hasOwnProperty("code")
+      ? (responseErrorData.code as string)
+      : "unknown";
+    const message = responseErrorData.hasOwnProperty("message")
+      ? (responseErrorData.message as string)
+      : "unknown";
+    const mappedError = ERRORS.getError(status, code);
+    if (mappedError) {
+      return mappedError.message || message;
```

Comment:
The `FeedbackValidationException` exception [bundles](https://github.com/ansible/ansible-wisdom-service/blob/main/ansible_wisdom/ai/api/views.py#L190) Django's serialiser validation messages into `message`.

The `ErrorHolder` [definition](https://github.com/ansible/vscode-ansible/pull/1165/files#diff-ad142e2781f46bf9b513b6d5d1d031dbe7ac71c1860a992310ccebea0e18ff5eR105) in VSCode does not include a message of its own and hence we use that which was returned from the service... We'd like to think `FeedbackValidationException` is never raised when using VSCode but I support server-provided messages should one not be defined in VSCode.